### PR TITLE
Update OpenStudio::DateTime to include a utc offset member and increa…

### PR DESCRIPTION
**Pull request by macumber**
…se support of ISO8601 formatted strings

### CODE REVIEW

**Comment by macumber** [comment]
@henryHorsey @nllong @jasondegraw  FYI

This pull request should not be merged lightly, I tried addressing places where time strings were written to databases and used for comparison but may not have gotten all of them

**Comment by brianlball** [comment]
is the calibration report part of the testing framework for this?  That uses the DateTime functionality

**Comment by macumber** [comment]
I don't think there are enough automated tests of the calibration report, but you're right that is a good test case

**Comment by macumber** [comment]
@kbenne can you look this over and merge for 1.11?

File: openstudiocore/src/project/ProjectDatabase.cpp [hunk 1]
```diff
@@ -186,6 +186,7 @@ namespace detail {
     // make sure we are the last one using database connection
     OS_ASSERT(m_qSqlDatabase.use_count() == 1);
 
+    QString connectionName = m_qSqlDatabase->connectionName();
```
Comment (macumber) [inline openstudiocore/src/project/ProjectDatabase.cpp]: @kbenne I was attempting to get rid of the error messages about closing QSqlDatabase connections still in use.  I did not solve this issue.

File: openstudiocore/src/project/Test/ProjectFixture.cpp [hunk 2]
```diff
@@ -88,7 +88,11 @@ openstudio::project::ProjectDatabase ProjectFixture::getDatabase(
 void ProjectFixture::removeAllDatabases() {
   for (const openstudio::path& dbDir : databaseDirs) {
     if (boost::filesystem::exists(dbDir)) {
-      boost::filesystem::remove_all(dbDir);
+      try{
```
Comment (macumber) [inline openstudiocore/src/project/Test/ProjectFixture.cpp]: This was failing and causing several tests to fail.  I did not understand why, but it was causing unit test failures

**Pull request merged**